### PR TITLE
4 pages within the "Work with Realm" section

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-12482

### Docs staging link (requires sign-in on MongoDB Corp SSO):
Initialize: https://docs-mongodbcom-staging.corp.mongodb.com/realm/caleb/workwithrealm/dotnet/init-realmclient.html
Email/Password: https://docs-mongodbcom-staging.corp.mongodb.com/realm/caleb/workwithrealm/dotnet/manage-email-password-users.html
APIKeys:https://docs-mongodbcom-staging.corp.mongodb.com/realm/caleb/workwithrealm/dotnet/create-manage-api-keys.html
Mulit-User: https://docs-mongodbcom-staging.corp.mongodb.com/realm/caleb/workwithrealm/dotnet/work-with-multiple-users.html

### CODE REVIEW

File: source/dotnet/manage-email-password-users.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.. _dotnet-manage-email-password-users:
+.. _android-manage-email-password-users:
```
Comment (cbush) [inline source/dotnet/manage-email-password-users.txt:1]: Why this change?

File: examples/dotnet/Examples/WorkWithRealm.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                {
+                    // disable the key
+                    await apiKeyClient.DisableAsync(key.Id);
+                }
+                //:code-block-end:
+            }
+            {
+                //:code-block-start:apikey-delete
+                var apiKeyClient = user.ApiKeys;
+                var key = await apiKeyClient.FetchAsync(ObjectId.Parse("00112233445566778899aabb"));
```
Comment (nirinchev) [inline examples/dotnet/Examples/WorkWithRealm.cs:111]: Seems a bit redundant to fetch the key only to delete it after that.

File: source/dotnet/work-with-multiple-users.txt [hunk 3]
```diff
@@ -37,7 +37,7 @@ User Account States
 When a user first logs in through a {+service-short+} SDK on a given device or
 browser, the SDK saves the user's information and keeps track of the user's
 state on the device. The user's data remains on the device, even if they log
```
Comment (nirinchev) [inline source/dotnet/work-with-multiple-users.txt:39]: This is a bit of a departure from other SDKs, but with .NET logging a user out removes their data.

File: source/dotnet/work-with-multiple-users.txt [hunk 4]
```diff
@@ -51,7 +51,7 @@ The following states describe an on-device user at any given time:
     more information.
    
   - **Inactive**: all authenticated users that are not the current
-    active user. You can :ref:`switch the active user <ios-switch-user>` to a
+    active user. You can :ref:`switch the active user <dotnet-switch-user>` to a
     currently inactive user at any time.
 
 - **Logged Out:** any user that authenticated on the device but
```
Comment (nirinchev) [inline source/dotnet/work-with-multiple-users.txt:57]: I can't comment on the correct line, but it's a bit weird to use `it` for user. Perhaps `or had *their* session revoked`?

File: source/dotnet/work-with-multiple-users.txt [hunk 5]
```diff
@@ -66,89 +66,48 @@ transition between states when certain events occur:
...
 
 .. _dotnet-switch-user:
 
 Switch the Active User
 ~~~~~~~~~~~~~~~~~~~~~~
 
-You can quickly switch an app's :ref:`active user <active-user>` to another
-logged in user at any time.
+You can switch an app's :ref:`active user <active-user>` to another
+logged in user bt calling
```
Comment (nirinchev) [inline source/dotnet/work-with-multiple-users.txt:96]:
```suggestion
logged in user by calling 
```

File: examples/dotnet/Examples/WorkWithRealm.cs [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                    Console.WriteLine($"User {user.Id} is logged on via {user.Provider}");
+                }
+                Assert.AreEqual(2, app.AllUsers.Count());
+                //:code-block-end:
+                //:code-block-start:multi-switch
+                app.SwitchUser(aimee);
+                if (app.CurrentUser.Id == aimee.Id)
+                {
+                    Console.WriteLine($"Aimee is now the current user.");
+                }
```
Comment (nirinchev) [inline examples/dotnet/Examples/WorkWithRealm.cs:150]: Perhaps use `Assert.IsTrue` similar to the other examples. Unless there's a bug in our code, this `if` block should always be `true` and we shouldn't imply otherwise.

File: examples/dotnet/Examples/WorkWithRealm.cs [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                //:code-block-end:
+                Assert.IsTrue(aimee.Id == app.CurrentUser.Id, "aimee is current user");
+
+                //:code-block-start:multi-remove
+                await app.RemoveUserAsync(elvis);
+                var noMoreElvis = app.AllUsers.FirstOrDefault(u => u.Id == elvis.Id);
+                if (noMoreElvis == null)
+                {
+                    Console.WriteLine($"Elvis has left the application.");
+                }
```
Comment (nirinchev) [inline examples/dotnet/Examples/WorkWithRealm.cs:160]: Same here - assert rather than use an if-check

File: source/dotnet/manage-email-password-users.txt (hunk 1 above)
Comment (MongoCaleb) [reply source/dotnet/manage-email-password-users.txt:1]: wait, what? It's `dotnet` in mine. Merge issue?

File: source/dotnet/init-realmclient.txt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+   {+app+}s or to the same {+app+} with different configurations. All
+   ``App`` instances that share the same App ID use the same underlying
+   connection.
+
+.. _dotnet-app-client-configuration:
+
+Configuration
+-------------
+For granular control of your app connection, such as custom timeouts for connections
+and keys for local encryption, you can pass an
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/dotnet/init-realmclient.txt:38]: The code example shows setting the timeout and log level, not encryption. The description just mentions its one of the things that *can* be in there... too confusing?

File: source/dotnet/manage-email-password-users.txt [hunk 9]
```diff
@@ -14,39 +14,78 @@ Manage Email/Password Users
...
+.. literalinclude:: /examples/generated/code/start/WorkWithRealm.codeblock.reset-user-1.cs
+   :language: csharp
+
+Password reset emails contain two values, ``token`` and ``tokenId``.
+To complete the password reset flow, you can reset the user's password on the client 
+or by calling a custom function on the backend.
+
+Reset the Password on the Client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To rewet the passworf on the client, your UI should prompt the user to enter a new
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/dotnet/manage-email-password-users.txt:68]: geez, who typed that?

File: source/dotnet/create-manage-api-keys.txt [hunk 10]
```diff
@@ -12,53 +12,41 @@ Create & Manage API Keys
    :depth: 2
    :class: singlecol
 
-Overview
---------
-
 .. _dotnet-api-key-create-user-key:
 
 Create a User API Key
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/dotnet/create-manage-api-keys.txt:17]: Intentional: ideally many of our pages would have less talk, more action. I know it differs from the others, but all the other languages have this simplicity for this page.

File: source/dotnet/work-with-multiple-users.txt (hunk 3 above)
Comment (MongoCaleb) [reply source/dotnet/work-with-multiple-users.txt:39]: @nirinchev  oh wow. That is a big difference. What's the reason for the departure?

File: source/dotnet/work-with-multiple-users.txt (hunk 4 above)
Comment (MongoCaleb) [reply source/dotnet/work-with-multiple-users.txt:57]: copy-pasta and I didn't catch that!

File: examples/dotnet/Examples/WorkWithRealm.cs (hunk 6 above)
Comment (MongoCaleb) [reply examples/dotnet/Examples/WorkWithRealm.cs:150]: @nirinchev  You've been opposed to using `Assert` in other code examples. OK here, then?

File: source/dotnet/init-realmclient.txt (hunk 8 above)
[the remark this answers is not in the document]
Comment (nirinchev) [reply source/dotnet/init-realmclient.txt]: Yeah, I'll drop mentioning the encryption key in this section because it requires a longer explanation.

File: source/dotnet/work-with-multiple-users.txt (hunk 3 above)
Comment (nirinchev) [reply source/dotnet/work-with-multiple-users.txt:39]: The implementation is wrong in the other SDKs. We are going to converge eventually and expose a way to log out a user without deleting their local data in the .NET SDK, but that's not going to make it for the beta.

File: source/dotnet/manage-email-password-users.txt [hunk 11]
```diff
@@ -14,39 +14,81 @@ Manage Email/Password Users
...
+your {+service-short+} ``App``'s ``EmailPasswordAuth`` instance:
 
-      .. code-block:: csharp
+.. admonition:: Use Deep Links in Mobile Apps
+   :class: note
+
+   Mobile applications can handle email confirmation directly in the app
+   by configuring `deep linking <https://developer.android.com/training/app-links/deep-linking.html>`_
+   in Android or `universal links <https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html>`_
+   in iOS.
```
Comment (cbush) [inline source/dotnet/manage-email-password-users.txt:49]: Is there a Xamarin wrapper for this?

File: source/includes/dotnet-initialize-realm.rst [hunk 12]
```diff
@@ -0,0 +1,54 @@
+Initialize Realm
+----------------
+
+Before you can use {+service-short+} in your app, you must
+initialize the {+service-short+} library. Your application should
+initialize {+service-short+} just once each time the application runs.
+
+To initialize the {+service-short+} library, provide an Android
```
Comment (cbush) [inline source/includes/dotnet-initialize-realm.rst:8]: This is still Android-specific?

**Review by cbush — APPROVED** [review]
Looks good pending search for all instances of "Android".

File: examples/dotnet/Examples/WorkWithRealm.cs (hunk 6 above)
Comment (nirinchev) [reply examples/dotnet/Examples/WorkWithRealm.cs]: Yeah, talk about mixed signals, right 😅 Generally, I'm against showing asserts, but in this context I think it's fine because:
1. It is consistent with the first two examples
2. The methods being called don't return a result and there's no meaningful way to demonstrate the effect of the method.

When the developer calls `app.SwitchUser(foo)`, there's nothing particularly interesting that they can do with `app.CurrentUser` that we can showcase to demonstrate that `CurrentUser == foo`.

File: source/includes/dotnet-initialize-realm.rst (hunk 12 above)
Comment (nirinchev) [reply source/includes/dotnet-initialize-realm.rst:8]: There's no initialization step for .NET, so probably can delete this file altogether.

File: source/dotnet/create-manage-api-keys.txt [hunk 13]
```diff
@@ -12,53 +12,41 @@ Create & Manage API Keys
...
 ---------------------
 
-.. tabs-realm-languages::
-   
-   .. tab::
-      :tabid: c-sharp
-
-      .. code-block:: csharp
+.. literalinclude:: /examples/generated/code/start/WorkWithRealm.codeblock.apikey-create.cs
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/dotnet/create-manage-api-keys.txt:20]: @nirinchev Guess what I just learned? The syntax highlighting library chokes with the `$"{foo}"`. :( So either we don't get highlighting or I use old-skool strings. :(

File: source/includes/dotnet-initialize-realm.rst (hunk 12 above)
Comment (MongoCaleb) [reply source/includes/dotnet-initialize-realm.rst:8]: That file was supposed to be deleted...

File: source/dotnet/create-manage-api-keys.txt (hunk 13 above)
[the remark this answers is not in the document]
Comment (nirinchev) [reply source/dotnet/create-manage-api-keys.txt:20]: Ewww... old school strings are so 2001 🤮 I don't know what's less bad - I guess we could keep using string interpolation and file an issue/ticket/fax to whoever maintains the syntax highlighting library and beg them to fix it. Then eventually, we'll only have to upgrade the library and that'll be it. The alternative is a bunch of work now to migrate to string.Format and then a bunch of work later to revert. But I'll trust your judgement here - I have no idea what our users will find more annoying.
